### ISSUE TITLE
better dependency in-notebook management

### ISSUE THREAD

**Issue by antonkulaga** [andypetrella/spark-notebook#801]
I've updated to the master and I noticed that you deleted the dependency command (BTW in Future it is better to show such warning in red as It took me some time to understand that it is not working anymore). Instead, you suggest using notebook-metadata file. I am not against it but I have to admit that having cells for dependencies was much more manageable: I saw the process of dependency download and got instant feedback if something went wrong. In case of notebook meta-data it is hard to notice what is happening there, wether the download process is still going on or if I can already use new deps. And if I make a type error in the dependency name I have to open those cluncky sidebar to see the error log instead of having errors in easier place.

**Comment by vidma** [comment]
well, that's true that UI could be better.

one of the reasons is that these `:dp, :cp, etc` commands were very ineffective - running it required restarting the whole REPL & spark-context. I doubt if all users were aware of it.

there might be some more advanced UI for notebook config & dependencies, and maybe also for investigating logs during "REPL boot". (P.S. I aggree, sidebar is cluncky, but the browser console logs are +/- fine IMO)

**Comment by antonkulaga** [comment]
>running it required restarting the whole REPL & spark-context. I doubt if all users were aware of it.

It is better to do this than to have what I have now: I deleted some dependencies in metadata file - nothing changed even after saving and reopening notebook. I added a custom repo + extra deps - got not errors/warning but they were not resolved. 
The format that you propose in metadata for custom repo is also strange (different from what I have in sbt) - I do not get what "defaul" means there.

**Comment by vidma** [comment]
yes, it would be nice to match the standard sbt dependency format!

**Comment by hblt-j** [comment]
Aha, is this internal discussion?
The first time the user is not aware of whether the notebook is smart to be able to load dependent jars directly to the process environment variables, and your example loaded aws jars, there is no prompt how to operate, restart set what. So the customer tried to load the mysql driver, found no work! The The
Now do not want him more perfect, but let him run up, what specific steps is, can you give the next (find spark-notebook documents all understanding ah), such as dep finished after the need to configure the context, put jars External dependencies jas directory, or restart, they are not in the process environment, people puzzled. The The
Of course, can be done under the shell can be added jar and update the best depend on the shell under the function I do not know how the progress, not a direct interface plus a restart button?
What's important now what do I do now?

**Comment by Johnny-P-Quarry** [comment]
@hblt-j  sorry, but what ?
